documentation/Using-the-SDK/Creating-your-plugin.md

```diff
@@ -1,69 +1,255 @@
 
-### Picking your SDK version
+#### Picking your SDK version
 The version of the SDK you add to your project will determine which versions of SDK drivers your plugin will work with. For example, a plugin that depends on SDK version 0.109.2 will not work a version of an SDK driver that uses SDK version 1.0.0.
 
 To decide which version of the SDK to use, refer to the [known SDK driver compatability lists](../Known-SDK-Driver-Compatibility/Overview.md).
 
-## Install WPA for Debugging
+#### Install WPA for Debugging
 
 One way to debug an SDK plugin project is to use WPA. Before we setup our project for this, WPA will need to be installed. 
 Please see [Using the SDK/Installing WPA](./Installing-WPA.md) for more information how to install WPA.
 
-## Setup for Debugging Using WPA
+#### Setup for Debugging Using WPA
 
 1) Right click your project and select "Properties"
 2) Select the "Debug" tab on the left
    * In newer versions of Visual Studio, you may need to click "Open debug launch profiles UI"
 3) For "Launch", select "Executable"
 4) For the "Executable", place the path to the `wpa.exe` that you previously installed as part of the WPT
    * Typically this might be: `C:\Program Files (x86)\Windows Kits\10\Windows Performance Toolkit\wpa.exe`
-5) For "Command line Arguments", add `-addsearchdir <bin folder for your plugin>` (e.g. `-addsearchdir C:\MyAddIn\bin\Debug\netstandard2.1`)
+5) For "Command line Arguments", add `-addsearchdir <bin folder for your plugin>` (e.g. `-nodefault -addsearchdir C:\MyAddIn\bin\Debug\netstandard2.1`)
+
+The above changes will cause Visual Studio to launch your installed WPA with the command line arguments listed above passed in whenever you start debugging your plugin (via `Debug` => `Start Debugging`). The commandline arguments listed above tell WPA to load plugins only from the directory provided.
+
+If you encounter issues loading your plugin in WPA, please refer to [Troubleshooting documentation](../Troubleshooting.md#wpa-related-issues).
+
+---
+
+## Creating a ProcessingSource
+
+Every plugin, regardless of the plugin framework chosen below, requires at least one `ProcessingSource`. Each `ProcessingSource` is an entry point for your plugin; they are what the SDK looks for when your plugin loads.
+
+### Create a ProcessingSource class
+
+In the project created above, replace the default `Program` class with one that extends the `ProcessingSource` class provided by the SDK.
+
+```cs
+public class MyProcessingSource : ProcessingSource
+{
+   public MyProcessingSource() : base()
+   {
+   }
+}
+```
+
+Each `ProcessingSource` is responsible for "discovering" the `Tables` that will be built. By default, a `ProcessingSource` will "discover" all `Tables` defined in its assembly. If you wish to override this behavior, refer to [Custom Table Discovery](./Advanced/Custom-Table-Discovery.md). _Most plugins do not need to override this behavior_.
```

Comment:
>Tables [](http://example.com/codeflow?start=62&length=6)

simple tables?

Reply:
I will move this into "creating-a-simple-sdk-plugin since discovery is only relevant there